plugins/actuators/vllm_performance/ado_actuators/vllm_performance/vllm_performance_test/execute_benchmark.py

```diff
@@ -149,14 +157,83 @@ def execute_random_benchmark(
     )
 
 
+def execute_geospatial_benchmark(
+    base_url: str,
+    model: str,
+    dataset: str,
+    num_prompts: int = 500,
+    request_rate: int | None = None,
+    max_concurrency: int | None = None,
+    hf_token: str | None = None,
+    benchmark_retries: int = 3,
+    retries_timeout: int = 5,
+    burstiness: float = 1,
+    interpreter: str = "python",
+) -> dict[str, Any]:
+    """
+    Execute benchmark with random dataset
+    :param base_url: url for vllm endpoint
+    :param model: model
+    :param data_set: data set name ["sharegpt", "sonnet", "random", "hf"]
+    :param hf_token: huggingface token
+    :param benchmark_retries: number of benchmark execution retries
+    :param retries_timeout: timeout between initial retry
+    :param input_token_length: length of input tokens
+    :param output_token_length: length of output tokens
+    :return: results dictionary
+    """
+
+    if dataset in default_geospatial_datasets_filenames:
+        from pathlib import Path
+
+        dataset_filename = default_geospatial_datasets_filenames[dataset]
+        parent_path = Path(__file__).parents[1].absolute()
+        data_set_path = os.path.join(parent_path, "datasets", dataset_filename)
```

Comment:
If you use Paths everywhere you can use the `/` operator to concatenate the paths in an os-independent way instead of using os.path